src/main/java/com/zachary_moore/runner/LintRunner.java

```diff
@@ -5,78 +5,74 @@
-                        singletonList(new JSONFile(basePathFile)));
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-        } else if (basePathFile.isDirectory()){
-            this.filesToLint = getAllFiles(basePath).stream()
-                    .map(f -> {
-                        try {
-                            return new JSONFile(f);
-                        } catch (IOException e) {
-                            throw new RuntimeException(e);
-                        }
-                    })
-                    .filter(Objects::nonNull)
-                    .collect(Collectors.toList());
-        } else {
-            throw new RuntimeException("Input path is neither a file nor directory");
-        }
+        this.basePaths = basePaths;
     }
 
-    private List<File> getAllFiles(String basePath) {
-        List<File> files = new ArrayList<>();
-        File directory = new File(basePath);
+    private Set<JSONFile> getAllFiles(String... basePaths) {
+        Set<File> files = new HashSet<>();
 
-        File[] fList = directory.listFiles();
-        if (fList != null) {
-            for (File file : fList) {
-                if (file.isFile()) {
-                    files.add(file);
-                } else if (file.isDirectory()) {
-                    files.addAll(getAllFiles(file.getAbsolutePath()));
-                }
+        for (String basePath : basePaths) {
+            File file = new File(basePath);
+
+            if (file.isFile()) {
+                files.add(file);
+            } else if (file.isDirectory()) {
+                files.addAll(FileUtils.listFiles(file, null, true));
+            } else {
+                // TODO: Add proper logging
+                Logger.getGlobal().warning(basePath + " is not a valid file path");
             }
         }
-        return files;
+
+        return files.stream().map(file -> {
+            try {
+                return new JSONFile(file);
+            } catch (IOException e) {
+                // Is not a JSON file
+                return null;
+            }
+        }).filter(Objects::nonNull).collect(Collectors.toSet());
     }
 
     /**
-     * Lint all files in given path in constructor
+     * Lint all files in given path in constructor and store the output
      * @return Representation of any lint issues
      */
     public Map<LintRule, Map<JSONFile, List<String>>> lint() {
         Map<LintRule, Map<JSONFile, List<String>>> lintOutput = new HashMap<>();
+        Set<JSONFile> filesToLint = getAllFiles(this.basePaths);
+
         for (LintRule lintRule : lintRegister.getLintRules()) {
             try {
                 if (lintRule.getLevel() != LintLevel.IGNORE) {
-                    Map<JSONFile, List<String>> lintReports = lintRule.lint(filesToLint.toArray(new JSONFile[filesToLint.size()]));
+                    Map<JSONFile, List<String>> lintReports = lintRule.lint(filesToLint.toArray(new JSONFile[0]));
```

Comment:
shouldn't this be `filesToLint.toArray(new JSONFile[<total elements in set>]`

Reply:
new JSONFile[0] is used only as a type hint. While there is no difference in the return value, using an empty array is the preferred approach because it runs faster on Hotspot.

https://stackoverflow.com/questions/174093/toarraynew-myclass0-or-toarraynew-myclassmylist-size